bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/SerialisedInputsToActionInputs.java

```diff
@@ -0,0 +1,131 @@
+            // We have to convert Double type into the target type.
+            try {
+                return switch (input.getType()) {
+                    case "byte", "java.lang.Byte" -> Byte.valueOf(valueDouble.byteValue());
+                    case "short", "java.lang.Short" -> Short.valueOf(valueDouble.shortValue());
+                    case "int", "java.lang.Integer" -> Integer.valueOf(valueDouble.intValue());
+                    case "long", "java.lang.Long" -> Long.valueOf(valueDouble.longValue());
+                    case "float", "java.lang.Float" -> Float.valueOf(valueDouble.floatValue());
+                    case "org.openhab.core.library.types.DecimalType" -> new DecimalType(valueDouble);
+                    default -> argument;
+                };
+            } catch (NumberFormatException e) {
+                LOGGER.warn(
+                        "Action {} input parameter '{}': converting value {} into type {} failed! Input parameter is ignored.",
+                        actionType.getUID(), input.getName(), argument, input.getType());
+                return null;
+            }
+        } else if (argument instanceof String valueString) {
+            // String value is accepted to instantiate few target types
+            try {
+                return switch (input.getType()) {
+                    case "boolean", "java.lang.Boolean" -> Boolean.valueOf(valueString.toLowerCase());
+                    case "byte", "java.lang.Byte" -> Byte.valueOf(valueString);
+                    case "short", "java.lang.Short" -> Short.valueOf(valueString);
+                    case "int", "java.lang.Integer" -> Integer.valueOf(valueString);
+                    case "long", "java.lang.Long" -> Long.valueOf(valueString);
+                    case "float", "java.lang.Float" -> Float.valueOf(valueString);
+                    case "double", "java.lang.Double" -> Double.valueOf(valueString);
+                    case "org.openhab.core.library.types.DecimalType" -> new DecimalType(valueString);
+                    case "java.time.LocalDate" ->
+                        // Accepted format is: 2007-12-03
+                        LocalDate.parse(valueString);
+                    case "java.time.LocalTime" ->
+                        // Accepted format is: 10:15:30
+                        LocalTime.parse(valueString);
+                    case "java.time.LocalDateTime" ->
+                        // Accepted format is: 2007-12-03T10:15:30
+                        // TODO documentation mention YYYY-MM-DD hh:mm
+                        LocalDateTime.parse(valueString);
+                    case "java.time.ZonedDateTime" ->
+                        // Accepted format is: 2007-12-03T10:15:30+01:00[Europe/Paris]
+                        ZonedDateTime.parse(valueString);
+                    case "org.openhab.core.library.types.QuantityType" -> new QuantityType<>(valueString);
```

Comment:
We also need to catch `IllegalArgumentException` for the QuantityType constructor.
I am also wondering if we should rather use the `QuantityType(String value, Locale locale)` constructor as this would allow the user to enter the input following his decimal separator (currently one has to use the decimal point as in the English locale).

Reply:
Note that your case will not work as the type will not be e this one. It will also include a dimension.

Reply:
We should certainly use startWith

Reply:
> Note that your case will not work as the type will not be this one. It will also include a dimension.

My case will work, as the UI will provide a string like `500 W` or `0.5 kW`.

Reply:
I was talking about the case on type. This case will never be considered.
I will fix that after testing.

Reply:
I don't understand what you mean, I have tested this code (or a previous version) and it worked.

Reply:
> We also need to catch `IllegalArgumentException` for the QuantityType constructor.

Already done.

> I am also wondering if we should rather use the `QuantityType(String value, Locale locale)` constructor as this would allow the user to enter the input following his decimal separator (currently one has to use the decimal point as in the English locale).

Not convinced. We should then consider locale also for any decimal.
In Main UI, do we accept a decimal value with a format depending on locale (when filling a configuration parameter) ?

Reply:
With the current implementation we don’t have to care about the decimal point … but I am not sure Main UI allows to use any decimal point — any way, as the UI validated the input the user will notice what’s correct.

Reply:
So, you are ok to ignore locale and let the code as it is ?

Reply:
With our current implementation we should use the constructor that accepts decimal and the unit, not the string constructor.

Reply:
My last commit changed that.
Now the only case where the QuantityType constructor with String parameter is used is when the passed value is a string that does not represent a number. This will never happen in your new UI as you will provide a number but it is accepted when using API explorer.